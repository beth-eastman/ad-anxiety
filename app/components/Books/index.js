/**
 * A resource page containing text about books.
 *
 * Bethany Eastman
 */
import React from 'react';
import { Card, CardText } from 'material-ui';
import { books } from 'data/resources';

const styles = {
  title: {
    textDecoration: 'underline',
  },
};

export default class Links extends React.Component { // eslint-disable-line

  /* Change the AppBar title */
  componentWillMount() {
    this.props.setPageTitle("Books"); // eslint-disable-line
    this.props.changeIcon('Back', '/resources/'); // eslint-disable-line
  }

  /* For each book, cite its title and description */
  render() {
    return (
      <Card>
        <CardText>
          {books.map((book) => (
            <div id="link" key={book.title}>
              <h4 style={styles.title}>{ book.title }</h4>
              { book.text }
              <br />
              { book.description }
              <br />
            </div>
          ))}
        </CardText>
      </Card>
    );
  }
}
