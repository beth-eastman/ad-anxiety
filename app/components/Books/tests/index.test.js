/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import Books from '../index';

it('renders Books', () => {
  const tree = renderer.create(
    <MuiThemeProvider>
      <Books setPageTitle={function () {}} changeIcon={function () {}} />
    </MuiThemeProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
