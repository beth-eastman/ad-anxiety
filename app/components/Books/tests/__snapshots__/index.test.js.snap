exports[`test renders Books 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }>
  <div
    style={
      Object {
        "paddingBottom": 8,
      }
    }>
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16,
        }
      }>
      <div
        id="link">
        <h4
          style={
            Object {
              "textDecoration": "underline",
            }
          }>
          Mastery of Your Anxiety and Worry: Workbook (Treatments That Work).
        </h4>
        by Craske, Michelle G. and Barlow, David H. 2006 Oxford University Press: New York
        <br />
        This is a workbook allowing you to work on exercises by two of the best know researchers in the field.
        <br />
      </div>
      <div
        id="link">
        <h4
          style={
            Object {
              "textDecoration": "underline",
            }
          }>
          Don\'t Panic: Taking Control of Anxiety Attacks. Third Edition.
        </h4>
        Wilson, Reid. 2009 Harper Collins Publishers: New York
        <br />
        A comprehensive self-help approach to treating Panic Disorder.
        <br />
      </div>
      <div
        id="link">
        <h4
          style={
            Object {
              "textDecoration": "underline",
            }
          }>
          Stop Obsessing: How to Overcome Your Obsessions and Compulsions (Revised Edition). Foa, Edna B and Wilson, Reid.
        </h4>
        2001 Bantam Paperbacks
        <br />
        This book provides concrete guidance for overcoming OCD.
        <br />
      </div>
      <div
        id="link">
        <h4
          style={
            Object {
              "textDecoration": "underline",
            }
          }>
          The OCD Workbook: Your Guide to Breaking Free from Obsessive-Compulsive Disorder.
        </h4>
        Hyman, Bruce M. and Pedrick, Cherry. 2005 New Harbinger Publications
        <br />
        A workbook format which allows you to work with your own symptoms of OCD.
        <br />
      </div>
      <div
        id="link">
        <h4
          style={
            Object {
              "textDecoration": "underline",
            }
          }>
          Dying of Embarrassment: Help for Social Anxiety & Phobia.
        </h4>
        Markway, Barbara G., Pollard, Alec C., Flynn, Teresa, and Carmin, Cheryl N. 1992. New Harbinger Publications
        <br />
        A well-written book providing guidance for coping with Social Phobia.
        <br />
      </div>
    </div>
  </div>
</div>
`;
